key.ui/src/main/java/de/uka/ilkd/key/gui/docking/DockingLayout.java

```diff
@@ -135,65 +136,112 @@ private void setLayout(String layout) {
 
-            protected SaveAction(MainWindow mainWindow) {
+            private SaveAction(MainWindow mainWindow) {
                 super(mainWindow);
                 setName("Save Layout");
+                setMenuPath("View.Layout");
             }
 
             @Override
             public void actionPerformed(ActionEvent e) {
-                String name = Objects.requireNonNull(comboLayouts.getSelectedItem()).toString();
-                mainWindow.getDockControl().save(name);
+                String layout = null;
+                var iter = layouts.getElements().asIterator();
+                while (iter.hasNext()) {
+                    var b = (JRadioButtonMenuItem) iter.next();
+                    if (b.getModel().isSelected()) {
+                        layout = b.getText();
+                        System.out.println("saving in " + layout);
+                        break;
+                    }
+                }
+                mainWindow.getDockControl().save(layout);
             }
         }
 
-        class LoadAction extends MainWindowAction {
+        final class LoadAction extends MainWindowAction {
             private static final long serialVersionUID = 3130337190207622893L;
 
-            protected LoadAction(MainWindow mainWindow) {
+            private LoadAction(MainWindow mainWindow) {
                 super(mainWindow);
                 setName("Load Layout");
+                setMenuPath("View.Layout");
             }
 
             @Override
             public void actionPerformed(ActionEvent e) {
-                setLayout(Objects.requireNonNull(comboLayouts.getSelectedItem()).toString());
+                String layout = null;
+                var iter = layouts.getElements().asIterator();
+                while (iter.hasNext()) {
+                    var b = (JRadioButtonMenuItem) iter.next();
+                    if (b.getModel().isSelected()) {
+                        layout = b.getText();
+                        break;
+                    }
+                }
+                setLayout(layout);
             }
         }
 
-        toolBar.add(new JLabel("Layouts: "));
-        for (String s : LAYOUT_NAMES) {
-            comboLayouts.addItem(s);
-        }
-        toolBar.add(comboLayouts);
-        toolBar.add(new LoadAction(mainWindow));
-        toolBar.add(new SaveAction(mainWindow));
-        toolBar.addSeparator();
-        toolBar.add(new ResetLayoutAction(mainWindow));
-        return toolBar;
-    }
+        actions.add(new LoadAction(mainWindow));
+        actions.add(new SaveAction(mainWindow));
+        actions.add(new ResetLayoutAction(mainWindow));
 
-    @Override
-    public List<Action> getMainMenuActions(MainWindow mainWindow) {
         ensureActions(mainWindow);
+
         return actions;
     }
 }
 
 
-class SaveLayoutAction extends MainWindowAction {
+final class SaveLayoutAction extends MainWindowAction {
```

Comment:
Previously they were added to the menu. The only thing special about them is that they load/save a specific layout. I can remove them and instead add shortcuts for the regular load/save actions.

Reply:
For reference, here are screenshots:
| current main | this PR |
|------|------------------|
|![image](https://github.com/KeYProject/key/assets/94171076/bd979e57-f819-45e6-b12c-001e45ba41cb)|![image](https://github.com/KeYProject/key/assets/94171076/97a9d9d0-bd1e-49e6-9fd6-29e56a100010)|

Personally, I think that the old interface (left) is clearer. For example, in the right one, it is unclear if the layout is already changed when changing the selection of the radio button or only after selecting and clicking "Load Layout". This was already a problem of the toolbar version. In addition, the old version can be used with keyboard shortcuts.

Either way, if you remove the toolbar buttons, the pair of classes can be removed (or merged with the other pair).

Reply:
The intention of the left one was to have the shortcut available. Some sort of Quick-Save/Quick-Load known from games. I am not sure, whether this was popular. 

I think this could be useful, if more and more views are added, and we also gain different perspective, e.g., one for classic logic, and one for code-centric.

Currently, I would tend to the left one, but without the additional menu level of Load/Save.

Reply:
I have updated the code to flatten the menu (see my other comment for a screenshot).